nmdc_automation/workflow_automation/models.py

```diff
@@ -63,20 +47,35 @@ def workflow_process_factory(record: Dict[str, Any]) -> Union[DataGeneration, Wo
     wfe = cls(**record)
     return wfe
 
+def _normalize_record(record: Dict[str, Any]) -> Dict[str, Any]:
+    """ Normalize the record by removing the _id field and converting the type field to a string """
+    record.pop("_id", None)
+    # for backwards compatibility strip Activity from the end of the type
+    record["type"] = record["type"].replace("Activity", "")
+    normalized_record = _strip_empty_values(record)
 
-def get_base_workflow_execution_keys() -> List[str]:
-    """ Return the keys that are common to all workflow executions """
-    keys = set()
-    for k, v in WorkflowExecution.__annotations__.items():
-        if k.startswith("_") or k.startswith("class_"):
-            continue
-        keys.add(k)
-    return list(keys)
-
-
-
-
-
+    # type-specific normalization
+    # add type to Mags Analysis mags_list records
+    if normalized_record["type"] == "nmdc:MagsAnalysis" and "mags_list" in normalized_record:
+        for i, mag in enumerate(normalized_record.get("mags_list", [])):
+            if not mag.get("type"):
+                # Update the original dictionary in the list
+                normalized_record["mags_list"][i]["type"] = "nmdc:MagBin"
+            if "num_tRNA" in mag:
```

Comment:
Why is there specific handling for num_tRNA?

Reply:
`num_t_rna` is the key in the current MagBin schema. `num_tRNA` is an older key and exists in records that we may need to process.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/262/commits/664d93a18b2dd4150d3f160e67d99def45258582

Reply:
great catch, ill talk to Chienchi.

Reply:
@mbthornton-lbl Chienchi fixed this and i bumped the version on config in this branch. so you can take out this additional mapping if you like.